### PR TITLE
Add Miniasm module

### LINKED ISSUE #825: new module: miniasm
I think it would be good to have a module for [miniasm](https://bioconda.github.io/recipes/miniasm/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [ ] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

nf-core/bacass is using a [local module](https://github.com/nf-core/bacass/blob/master/modules/local/miniasm.nf)

This is also mentioned in nf-core/bacass#61

It might be related to [dragonflye](https://github.com/rpetit3/dragonflye/blob/main/README.md) which has a [nf-core module](https://github.com/nf-core/modules/tree/master/modules/dragonflye), issue #632 

_Be aware, that test data is not complete, the required `.paf` is missing currently. Bacteroides fragilis test-datasets seem to be not compatible._

### PR DESCRIPTION
**Pull request by avantonder**
New miniasm module added to nf-core/modules

## PR checklist

Closes #825  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/miniasm/meta.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  - assembly
+  - pacbio
+  - nanopore
+tools:
+  - miniasm:
+      description: Ultrafast de novo assembly for long noisy reads (though having no consensus step)
+      homepage: https://github.com/lh3/miniasm
+      documentation: https://github.com/lh3/miniasm
+      tool_dev_url: https://github.com/lh3/miniasm
+      doi: ""
```
Comment (d4straub) [inline modules/miniasm/meta.yml:13]: Found one more point: Please add the citation
```suggestion
      doi: "https://doi.org/10.1093/bioinformatics/btw152"
```

**Comment by lassefolkersen** [comment]
I looked this one through in detail, and ran the tests locally too. Also checked with the documentation at [miniasm-github](https://github.com/lh3/miniasm), and it seems it covers a reasonable use case and the output looks as expected too (apart from md5sums matching, obviously).  It's short and to-the-point as well  - all in all, I can't see any problems with this.

**Comment by sguizard** [comment]
I agree too ;)
Just have to fix the conda failed test.

**Review by d4straub — APPROVED** [review]
LGTM!

File: modules/miniasm/meta.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      description: File containing software versions
+      pattern: "versions.yml"
+  - gfa:
+      type: file
+      description: Assembly graph
+      pattern: "*.gfa.gz"
+  - assembly:
+      type: file
+      description: Genome assembly
+      pattern: "*_assembly.fasta.gz"
```
Comment (drpatelh) [inline modules/miniasm/meta.yml:48]:
```suggestion
      pattern: "*.fasta.gz"
```

File: tests/modules/miniasm/test.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+- name: miniasm test_miniasm
+  command: nextflow run tests/modules/miniasm -entry test_miniasm -c tests/config/nextflow.config
+  tags:
+    - miniasm
+  files:
+    - path: output/miniasm/test.gfa.gz
+      md5sum: c68e4c2b64338d1c0f5b79b32934da14
+    - path: output/miniasm/test_assembly.fasta.gz
```
Comment (drpatelh) [inline tests/modules/miniasm/test.yml:8]:
```suggestion
    - path: output/miniasm/test.assembly.fasta.gz
```

File: tests/modules/miniasm/main.nf [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { MINIASM } from '../../../modules/miniasm/main.nf' addParams( options: [:] )
```
Comment (drpatelh) [inline tests/modules/miniasm/main.nf:5]:
```suggestion
include { MINIASM } from '../../../modules/miniasm/main.nf' addParams( options: [suffix:'.assembly'] )
```

File: tests/modules/miniasm/test.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+- name: miniasm test_miniasm
+  command: nextflow run tests/modules/miniasm -entry test_miniasm -c tests/config/nextflow.config
+  tags:
+    - miniasm
+  files:
+    - path: output/miniasm/test.gfa.gz
```
Comment (drpatelh) [inline tests/modules/miniasm/test.yml:6]:
```suggestion
    - path: output/miniasm/test.assembly.gfa.gz
```

**Pull request merged**